superduperdb/db/base/data_backend.py

```diff
@@ -26,14 +25,24 @@ def drop(self, force: bool = False):
         """
         pass
 
-    @abstractmethod
-    def get_output_from_document(self, r: Document, key: str, model: str):
-        pass
-
+    # TODO - not the right place for this
     @abstractmethod
     def set_content_bytes(self, r, key, bytes_):
         pass
 
+    # TODO - not the right place for this
     @abstractmethod
     def unset_outputs(self, info):
         pass
+
+    def create_vector_index(self, vector_index):
```

Comment:
I presume that these methods are not `@abstractmethod`s because they are not always required? In which case, do we really need to include them here at all? What breaks if we do not include them.